tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/microprofiile/template/ConfigPropertyTestTemplate.java

```diff
@@ -0,0 +1,162 @@
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributor:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.cdi.bot.test.microprofiile.template;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.reddeer.common.wait.TimePeriod;
+import org.eclipse.reddeer.common.wait.WaitUntil;
+import org.eclipse.reddeer.common.wait.WaitWhile;
+import org.eclipse.reddeer.core.exception.CoreLayerException;
+import org.eclipse.reddeer.eclipse.condition.ProjectExists;
+import org.eclipse.reddeer.eclipse.jdt.ui.packageview.PackageExplorerPart;
+import org.eclipse.reddeer.eclipse.ui.views.markers.ProblemsView;
+import org.eclipse.reddeer.eclipse.ui.views.markers.ProblemsView.ProblemType;
+import org.eclipse.reddeer.swt.condition.ControlIsEnabled;
+import org.eclipse.reddeer.swt.condition.ShellIsAvailable;
+import org.eclipse.reddeer.swt.impl.button.PushButton;
+import org.eclipse.reddeer.swt.impl.combo.LabeledCombo;
+import org.eclipse.reddeer.swt.impl.ctab.DefaultCTabItem;
+import org.eclipse.reddeer.swt.impl.group.DefaultGroup;
+import org.eclipse.reddeer.swt.impl.menu.ContextMenuItem;
+import org.eclipse.reddeer.swt.impl.menu.ShellMenuItem;
+import org.eclipse.reddeer.swt.impl.shell.DefaultShell;
+import org.eclipse.reddeer.swt.impl.text.LabeledText;
+import org.eclipse.reddeer.swt.impl.tree.DefaultTreeItem;
+import org.eclipse.reddeer.workbench.condition.EditorHasValidationMarkers;
+import org.eclipse.reddeer.workbench.core.condition.JobIsRunning;
+import org.eclipse.reddeer.workbench.impl.editor.DefaultEditor;
+import org.eclipse.reddeer.workbench.impl.editor.TextEditor;
+import org.jboss.tools.cdi.bot.test.CDITestBase;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * 
+ * @author zcervink@redhat.com
+ * 
+ */
+public class ConfigPropertyTestTemplate extends CDITestBase {
+
+	@Before
+	public void prepareProject() {
+		PackageExplorerPart pexplorer = new PackageExplorerPart();
+		pexplorer.open();
+		pexplorer.getProject(PROJECT_NAME).select();
+
+		beansHelper.createClass(CDI_BEAN_1_JAVA_FILE_NAME, PACKAGE_NAME);
+		editResourceUtil.replaceClassContentByResource(CDI_BEAN_1_JAVA_FILE_NAME + JAVA_FILE_EXTENSION,
+				readFile("resources/configProperty/CdiBean1.jav_"), false, false);
+		TextEditor te = new TextEditor(CDI_BEAN_1_JAVA_FILE_NAME + JAVA_FILE_EXTENSION);
+		new WaitWhile(new EditorHasValidationMarkers(te));
+		te.save();
+		new WaitWhile(new EditorHasValidationMarkers(te));
+
+		convertToMavenProject(PROJECT_NAME, "war", false);
+		addDependency(PROJECT_NAME, "org.eclipse.microprofile.config", "microprofile-config-api", "2.0");
+		createApplicationPropertiesFile(PROJECT_NAME);
+	}
+
+	@Test
+	public void isConfigPropertyValidationAllowedTest() {
```

Comment:
I would also like to see a negative test case, meaning that without adding proper dependencies/imports, it should show a proper error.

Reply:
I have added negative testcase to test whether the Problems View shows right error messages when no Microprofile dependency is added to the pom.xml. I have included it into the current test because creating 2 tests means make them run is specific order (first check for errors, then add dependency + check again). @odockal is it ok to make one test for 2 testcases?

Reply:
sounds ok.

Reply:
And change the test case name to something more readable, ie. testMyNewFeature. thanks

Reply:
Renamed.